test/stream-management.test.js

```diff
@@ -93,6 +93,37 @@ test(
 
     xmpp = client({ credentials, service: domain });
     xmpp.streamManagement.timeout = 10;
+    xmpp.streamManagement.requestAckInterval = 5;
+    xmpp.streamManagement.debounceAckRequest = 1;
+    debug(xmpp);
+
+    const promise_resumed = promise(xmpp.streamManagement, "resumed");
+    await xmpp.start();
+    xmpp.send(
+      <iq to={domain} id="ping" type="get">
+        <ping xmlns="urn:xmppp:ping" />
+      </iq>,
+    );
+
+    // Pretend we don't receive the ack by removing event listeners
+    // on the socket
+    xmpp._detachSocket();
+
+    await promise_resumed;
+    expect().pass();
```

Comment:
this code reflects the test title properly `client reconnects when server fails to ack stanza`:

---

```
xmpp._detachSocket()
```

along with the comment clearly stats that it mimicks not receiving an ack

---

```
await promise_resumed;
expect().pass();
```

we are waiting for the stream to resume (= client reconnects) and pass the test

it proves the test right